pallets/dapp-staking-v3/src/test/mock.rs

```diff
@@ -179,21 +193,43 @@ pub(crate) fn _run_to_block(n: u64) {
 /// Run for the specified number of blocks.
 /// Function assumes first block has been initialized.
 pub(crate) fn run_for_blocks(n: u64) {
-    _run_to_block(System::block_number() + n);
+    run_to_block(System::block_number() + n);
 }
 
 /// Advance blocks until the specified era has been reached.
 ///
 /// Function has no effect if era is already passed.
 pub(crate) fn advance_to_era(era: EraNumber) {
-    // TODO: Properly implement this later when additional logic has been implemented
-    ActiveProtocolState::<Test>::mutate(|state| state.era = era);
+    assert!(era >= ActiveProtocolState::<Test>::get().era);
+    while ActiveProtocolState::<Test>::get().era < era {
+        run_for_blocks(1);
+    }
+}
+
+/// Advance blocks until next era has been reached.
+pub(crate) fn advance_to_next_era() {
+    advance_to_era(ActiveProtocolState::<Test>::get().era + 1);
 }
 
 /// Advance blocks until the specified period has been reached.
 ///
 /// Function has no effect if period is already passed.
 pub(crate) fn advance_to_period(period: PeriodNumber) {
-    // TODO: Properly implement this later when additional logic has been implemented
-    ActiveProtocolState::<Test>::mutate(|state| state.period = period);
+    assert!(period >= ActiveProtocolState::<Test>::get().period_number());
+    while ActiveProtocolState::<Test>::get().period_number() < period {
+        run_for_blocks(1);
+    }
+}
+
+/// Advance blocks until next period has been reached.
+pub(crate) fn advance_to_next_period() {
+    advance_to_period(ActiveProtocolState::<Test>::get().period_number() + 1);
+}
+
+/// Advance blocks until next period type has been reached.
+pub(crate) fn _advance_to_next_period_type() {
+    let period_type = ActiveProtocolState::<Test>::get().period_type();
+    while ActiveProtocolState::<Test>::get().period_type() == period_type {
+        run_for_blocks(1);
+    }
 }
```

Comment:
Why do we need this pub function here? I didn't see any references for it.

Reply:
I used it for one test, but then changed it later. If I don't use it by when I'm done implementing everything, I'll remove it.